src/GraphQL/Types/Schema.cs

```diff
@@ -345,11 +297,23 @@ private void CreateSchemaTypes()
                 types,
                 Directives,
                 type => (IGraphType)_services.GetRequiredService(type),
-                NameConverter);
+                this);
 
             // At this point, Initialized will return false, and Initialize will still lock while waiting for initialization to complete.
             // However, AllTypes and similar properties will return a reference to SchemaTypes without waiting for a lock.
             _allTypes.ApplyMiddleware(FieldMiddleware);
+
+            Validate();
+        }
+
+        /// <summary>
+        /// Validates correctness of the created schema. This method is called only once - during schema initialization.
+        /// </summary>
+        protected virtual void Validate()
+        {
+            //TODO: add different validations, also see SchemaBuilder.Validate
+            //TODO: checks for parsed SDL may be expanded in the future, see https://github.com/graphql/graphql-spec/issues/653
+            this.Run(new AppliedDirectivesVisitor(this));
```

Comment:
This line causes additional build time (and allocations?).

Reply:
Your last benchmark only shows 4% memory increase, and faster build time.  It's not a big deal for those very few people that have scoped schemas.  If they were concerned about speed or memory they would not use a scoped schema.

Reply:
Nevertheless, I have already taken this line under the condition.